src/Microsoft.AspNetCore.Razor.Evolution/ITagHelperBoundAttributeDescriptorBuilder.cs

```diff
@@ -0,0 +1,260 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    public sealed class ITagHelperBoundAttributeDescriptorBuilder
```

Comment:
You could also consider making this a nested class/interface in `BoundAttributeDescriptor` so that it reads like `BoundAttributeDescriptor.IBuilder`. 

The nice thing about this as it stands is that only the `Create` method is what ties it to an `ITagHelper`-based implementation.